modules/branch-keystore-node/src/branch_keystore.ts

```diff
@@ -0,0 +1,696 @@
+      encryptedBranchKey.type instanceof HierarchicalSymmetricVersion,
+      'Unexpected type. Not a version record.'
+    )
+
+    //= aws-encryption-sdk-specification/framework/branch-key-store.md#getbranchkeyversion
+    //# GetBranchKeyVersion MUST verify that the returned EncryptedHierarchicalKey MUST have a logical table name equal to the configured logical table name.
+    needs(
+      encryptedBranchKey.encryptionContext[TABLE_FIELD] ==
+        this.logicalKeyStoreName,
+      'Unexpected logical table name. Expected ${this.logicalKeyStoreName}, found ${encryptedBranchKey.encryptionContext[TABLE_FIELD}.'
+    )
+
+    //= aws-encryption-sdk-specification/framework/branch-key-store.md#getbranchkeyversion
+    //# If the branch key fails to decrypt, this operation MUST fail.
+
+    //= aws-encryption-sdk-specification/framework/branch-key-store.md#getbranchkeyversion
+    //# The operation MUST decrypt the branch key according to the [AWS KMS Branch Key Decryption](#aws-kms-branch-key-decryption) section.
+    const branchKey = await decryptBranchKey(this, encryptedBranchKey)
+
+    //= aws-encryption-sdk-specification/framework/branch-key-store.md#getbranchkeyversion
+    //# This GetBranchKeyVersion MUST construct [branch key materials](./structures.md#branch-key-materials)
+    //# according to [Branch Key Materials From Authenticated Encryption Context](#branch-key-materials-from-authenticated-encryption-context).
+    const branchKeyMaterials = constructBranchKeyMaterials(
+      branchKey,
+      encryptedBranchKey
+    )
+
+    //= aws-encryption-sdk-specification/framework/branch-key-store.md#getbranchkeyversion
+    //# This operation MUST return the constructed [branch key materials](./structures.md#branch-key-materials).
+    return branchKeyMaterials
+  }
+
+  //= aws-encryption-sdk-specification/framework/branch-key-store.md#getkeystoreinfo
+  //= type=implication
+  //# This operation MUST return the keystore information in this keystore configuration.
+  getKeyStoreInfo(): KeyStoreInfoOutput {
+    return {
+      keystoreId: this.keyStoreId,
+      //= aws-encryption-sdk-specification/framework/branch-key-store.md#getkeystoreinfo
+      //= type=implication
+      //# The [keystore name](#table-name) MUST be obtained
+      //# from the configured [KeyStorage](./key-store/key-storage.md#interface)
+      //# by calling [GetKeyStorageInfo](./key-store/key-storage.md#getkeystorageinfo).
+      keystoreTableName: this.storage.getKeyStorageInfo().name,
+      logicalKeyStoreName: this.logicalKeyStoreName,
+      grantTokens: this.grantTokens ? this.grantTokens.slice() : [],
+      kmsConfiguration: this.kmsConfiguration._config,
+    }
+  }
+}
+
+immutableClass(BranchKeyStoreNode)
+
+// type guard
+export function isIBranchKeyStoreNode(
+  keyStore: any
+): keyStore is BranchKeyStoreNode {
+  return keyStore instanceof BranchKeyStoreNode
+}
+
+// The JS implementation is not encumbered with the legacy construction
```

Comment:
Question: why do we need to put this here?

I thought we put duvet like
// comments from spec
Implementation

Reply:
All this is here to document what is and is not supported. So this is intended as the justification for why everything after this is an exception.